src/shared/sam/cli/samCliBuildCommand.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { extensionSettingsPrefix } from '../../constants'
+import { fileExists } from '../../filesystemUtilities'
+import { DefaultSettingsConfiguration } from '../../settingsConfiguration'
+import { ChildProcess, ChildProcessResult } from '../../utilities/childProcess'
+import { SamCliCommand } from './samCliCommand'
+import { SamCliConfiguration } from './samCliConfiguration'
+import { DefaultSamCliLocationProvider } from './samCliLocator'
+
+export interface SamCliBuildCommandResponse {
+}
+
+export class SamCliBuildCommand extends SamCliCommand<SamCliBuildCommandResponse> {
```

Comment:
Nit: Can we call these something like `SamCliCommandInvocation` and `SamCliBuildCommandInvocation`, to differentiate them from the commands themselves?

Reply:
I didn't understand what you mean here.
Currently `sam xxx` -> SamCliXxxCommand
`sam info` -> `SamCliInfoCommand`
`sam local invoke` -> `SamCliLocalInvokeCommand`
and this one is `sam build`

Reply:
A 'command' is like a function: A reusable action that can be invoked multiple times with different parameters.

These types don't represent commands, they represent invocations of commands, with the command's parameters specified in the constructor.

For example, in the following code, the arrow function is a command, and `GreetInvocation` is an invocation of that command.

```typescript
vscode.commands.registerCommand(
    'greet',
    async (name: string) => await new GreetInvocation(name).execute()
)

...

class GreetInvocation {
    public constructor(private readonly name: string) {
    }

    public async execute(): Promise<void> {
        console.log(`Hello, ${this.name}!`)
    }
}
```